python/ray/serve/async_goal_manager.py

```diff
@@ -8,34 +8,52 @@
 
 
 class AsyncGoalManager:
+    """
+    Essentially a dictionary of <goal_id, asyncio.Future> within
+    ServeController's event loop with APIs to facilitate common calls.
+    """
+
```

Comment:
Please add details about the failure handling logic here. What is the behavior and what are the guarantees if a goal is errored? How should the client handle this?

Reply:
ping